stageleft/src/runtime_support.rs

```diff
@@ -4,6 +4,27 @@ use std::mem::MaybeUninit;
 use proc_macro2::{Span, TokenStream};
 use quote::quote;
 
+pub fn get_final_crate_name(crate_name: &str) -> TokenStream {
+    let final_crate = proc_macro_crate::crate_name(crate_name)
+        .unwrap_or_else(|_| panic!("{crate_name} should be present in `Cargo.toml`"));
+
+    match final_crate {
+        proc_macro_crate::FoundCrate::Itself => {
+            if std::env::var("CARGO_BIN_NAME").is_ok() {
+                let underscored = crate_name.replace('-', "_");
+                let underscored_ident = syn::Ident::new(&underscored, Span::call_site());
+                quote! { #underscored_ident }
+            } else {
+                quote! { crate }
+            }
+        }
+        proc_macro_crate::FoundCrate::Name(name) => {
+            let ident = syn::Ident::new(&name, Span::call_site());
```

Comment:
was there before, but presumably these Span::call_site should get swapped for meaningful spans some day :tm:?

Reply:
Yeah... basically all the spans involved here are cross-crate spans, so some day ™️ indeed.